### DISCUSSION TITLE
How to mark an existing function as a workout without rewriting it?

### DISCUSSION THREAD

**Discussion by hoebat** [software-mansion/react-native-reanimated#2039] [Q&A]
I'd like to use some functionality of https://github.com/facebook/react-native/blob/master/Libraries/Utilities/MatrixMath.js to do some transformations in the UI thread.

However, I'm getting the error of 'Tried to synchronously call function from a different thread'

Is there a way of wrapping these functions in a form so that they are able to be used as a worklet?

**Comment by jgonet** [comment]
You can do it by utilizing `createWorklet` function we export.

**Comment by hoebat** [comment]
@jakub-gonet: can you give an example of how I can use this function?

**Reply by jgonet** [reply]
`const fAsWorklet = createWorklet(f);`
